packages/collabswarm/src/collabswarm-document.ts

```diff
@@ -466,7 +491,10 @@ export class CollabswarmDocument<
     this._hashes.add(hash);
 
     // Send new message.
-    const updateMessage = this._crdtProvider.newMessage(this.documentPath);
+    const updateMessage: CRDTSyncMessage<ChangesType> = {
```

Comment:
Oh is CRDTSyncMessages just 'defined' here as a type interface to ChangesType? There's this abstraction problem about what level to use to give an example I guess, but I do think examples are helpful if possible

Reply:
I think the right way to describe what's going on here is:

A new constant called `updateMessage` whose type must match `CRDTSyncMessage<ChangesType>` (ChangesType is a type parameter of the CollabswarmDocument that's being passed through to the CRDTSyncMessage ChangesType parameter. See: `packages/collabswarm/src/crdt-sync-message.ts`) is being initialized here to a new javascript object with some fields.

What kind of example are you thinking here? The CRDTSyncMessage is never handled by the end-user, this is purely an internal type used for communication between Collabswarm nodes.

Reply:
Great. "Example" refers to the above comment where we saw it's covered in the interface def so good to go here.